### PR TITLE
Additional support for PicoScope 4444

### PR DESCRIPTION
**Pull request by rascustoms**
Ported fix for USB power from ps5000a to ps4000a.
Enabled Bandwidth Filtering for PicoScope 4444

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Looks good. Can you please fix the flake8 and pep8 errors. Since we have no other tests, these kinda act as a minimum barrier of protection.

File: picoscope/picobase.py [hunk 1]
```diff
@@ -247,10 +247,12 @@ def setChannel(self, channel='A', coupling="AC", VRange=2.0,
         if not isinstance(BWLimited, int):
             BWLimited = self.BW_LIMITS[BWLimited]
 
-        if BWLimited == 2:
```
Comment (hmaarrfk) [inline picoscope/picobase.py]: we should just raise an error if the BWLimit is not in the the values of the dictionary above. That can be left as an issue for an other time.

Comment (hmaarrfk) [reply picoscope/picobase.py]: or use something like a default dict, to get BWLimit 0.

Comment (hmaarrfk) [reply picoscope/picobase.py]: don't  worry about addressing this now, it is more for future reference.

**Comment by hmaarrfk** [comment]
i guess there are some new flake8 errors that poped up recently. fixing them in https://github.com/colinoflynn/pico-python/pull/152

**Comment by rascustoms** [comment]
No problems on the pep, I'll have it done shortly. Would it be worth also addressing #58 in fixing the power settings? Could we add another default value to the class (not unlike addressing the bandwidth options) as to what behaviour the user wants?

**Comment by hmaarrfk** [comment]
I think maybe the correct fix seems to move:

```
        # This will check if the power supply is not connected
        # and change the power supply accordingly
        # Personally (me = Mark), I don't like this
        # since the user should address this immediately, and we
        # shouldn't let this go as a soft error
        # but I think this should do for now
        if m == 0x11A:
            self.changePowerSource(m)
        else:
            self.checkResult(m)

        self.model = self.getUnitInfo('VariantInfo')
```

To the high level wrapper.

We can then add a parameter: 'power_source' that would be `None`, `DC`, or `USB` where None would enable some magic to happen, while DC and USB would allow the user to specify exactly the source they desire.

File: picoscope/picobase.py (hunk 1 above)
Comment (rascustoms) [reply picoscope/picobase.py]: Should we refactor it so the values of that dict come from the device classes rather then the base class. As it seems the enum values (dict) have different meanings device to device? Reading the documentation for the 4444 it also appears that vary depending on what probes are attached. I.e. I get an error if I use the 20kHz setting with any of the probes I have for the 4444, but I assume there must be a probe that does support it, or why would it be in the API.

**Comment by hmaarrfk** [comment]
again, I have a philosophy that I want this to work for **you** the contributor, and for now, this isn't a blocker for me. If it is a blocker for you, then we can work to address it.

**Comment by hmaarrfk** [comment]
I think the refactor should just happen elsewhere. It will also require careful consideration. If it does depend on the device class, we can add secondary conditions to the if statements

**Pull request merged**